batch/batch/front_end/front_end.py

```diff
@@ -484,6 +504,17 @@
                 else:
                     state = 'Pending'
 
+                spec_writer.add(json.dumps(spec))
+
+                if batch_format_version > 1:
+                    spec = {
+                        'secrets': spec.get('secrets'),
+                        'service_account': spec.get('service_account'),
+                        'input_files': len(spec.get('input_files', [])) > 0,
+                        'output_files': len(spec.get('output_files', [])) > 0,
```

Comment:
I might suggest giving these different names if you're going to change the type.  'has_input_files', etc.